test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetConsoleTestCase.cs

```diff
@@ -969,6 +969,91 @@ public void VerifyInitScriptsExecution(ProjectTemplate projectTemplate)
+        {
+            EnsureVisualStudioHost();
+            using (var simpleTestPathContext = new SimpleTestPathContext())
+            {
+                // Arrange
+                var PackageName = "TestPackage";
+                var v100 = "1.0.0";
+                await CommonUtility.CreatePackageInSourceAsync(simpleTestPathContext.PackageSource, PackageName, v100);
+
+                using (var testContext = new ApexTestContext(VisualStudio, projectTemplate, Logger, addNetStandardFeeds: true, simpleTestPathContext: simpleTestPathContext))
+                {
+                    SolutionService solutionService = VisualStudio.Get<SolutionService>();
+                    var nugetConsole = GetConsole(testContext.Project);
+
+                    // Act
+                    nugetConsole.Execute($"find-package {PackageName} -ExactMatch");
+
+                    // Assert
+                    Assert.IsTrue(nugetConsole.IsMessageFoundInPMC(PackageName), $"The package name {PackageName} doesn't show correctly in PMC.");
+                    Assert.IsTrue(nugetConsole.IsMessageFoundInPMC(v100), $"The package version {v100} doesn't show correctly in PMC.");
+                }
+            }
+        }
+
+        [DataTestMethod]
+        [DynamicData(nameof(GetNetCoreTemplates), DynamicDataSourceType.Method)]
+        [Timeout(DefaultTimeout)]
+        public async Task VerifyCmdGetPackageUpdateInPMC(ProjectTemplate projectTemplate)
+        {
+            EnsureVisualStudioHost();
+            using (var simpleTestPathContext = new SimpleTestPathContext())
+            {
+                // Arrange
+                var packageName = "TestPackage";
+                var v100 = "1.0.0";
+                var v200 = "2.0.0";
+                await CommonUtility.CreatePackageInSourceAsync(simpleTestPathContext.PackageSource, packageName, v100);
+                await CommonUtility.CreatePackageInSourceAsync(simpleTestPathContext.PackageSource, packageName, v200);
+
+                using (var testContext = new ApexTestContext(VisualStudio, projectTemplate, Logger, addNetStandardFeeds: true, simpleTestPathContext: simpleTestPathContext))
+                {
+                    // Arrange
+                    SolutionService solutionService = VisualStudio.Get<SolutionService>();
+                    var nugetConsole = GetConsole(testContext.Project);
+
+                    nugetConsole.InstallPackageFromPMC(packageName, v100);
+                    testContext.SolutionService.Build();
+                    testContext.NuGetApexTestService.WaitForAutoRestore();
+                    nugetConsole.Clear();
+
+                    // Act
+                    nugetConsole.Execute("get-package -update");
+
+                    // Assert
+                    Assert.IsTrue(nugetConsole.IsMessageFoundInPMC(v200), $"The latest package version {v200} doesn't show correctly in PMC.");
```

Comment:
Same here. Run GetText and then use similar as in https://github.com/NuGet/NuGet.Client/pull/5426/files#diff-d40c827f8805190cb124af37c978b3ac4d843091e7d43a5500182596cf476640R75. 

Use something like. 
```cs
string PMCText = nugetConsole.GetText();
PMCText.Should().Contain("whatever the expected message is");
```

If this test fails, the then fail message will contain actionable information + you don't need to write a custom message. 

In general you want to see the `expected` and `actual` in the error message for test cases.

Reply:
👍